enhancements/workload-partitioning/rationalizing-configuration.md

```diff
@@ -0,0 +1,478 @@
+configuring the workload partitioning in a running cluster, so no
+controller is needed, for now.
+
+We propose that the API be owned by core OpenShift, and defined in the
+`openshift/api` repository. A separate enhancement will be written to
+describe the API in more detail.
+
+### Owner for Enablement
+
+Workload management needs two node configuration steps which overlap
+with work PAO is doing today:
+
+* Set reserved CPUs in kubelet config and enable the CPU manager
+  static policy
+* Set systemd affinity to match the reserved CPU set
+
+PAO already has logic for managing CPU pools and configuring kubelet
+and CRI-O. Since it already does a lot of what is needed to enable
+workload partitioning, we propose to extend it to handle the
+additional configuration currently being done manually by users,
+including the machine config with settings for kubelet and CRI-O.
+
+The effect of this decision is that the easiest way to use the
+workload partitioning feature will be through the
+performance-addon-operator. This means that the implementation of the
+feature is split across a few components, which avoids duplicating a
+lot of the work already done in the PAO.
+
+Workload partitioning is currently enabled as part of installing a
+cluster, and that will not change as part of the implementation of
+this design. Enabling partitioning early in this way provides
+predictable behavior from the beginning of the life-cycle, and avoids
+extra reboots required to enable it in a running cluster (an important
+consideration in environments with short, fixed maintenance
+windows). To continue to support enabling the feature during
+deployment, and to simplify the configuration, we propose to extend
+the PAO with a `render` command, similar to the one in other
+operators, to have it generate manifests for the OpenShift installer
+to consume. This avoids any need to change the OpenShift installer to
+support this uncommon use case.
+
+The `render` command will have an option to enable the management
+partition. When the option is given, the `render` command will create
+a manifest for the new enablement API with status data that includes
+the `management` workload type to enable partitioning (the details of
+that API will be defined in a separate design document). In the
+future, the PAO might enable partitioning for all of the types
+mentioned in the input PerformanceProfiles, or based on some other
+input.
+
+For each PerformanceProfile, the PAO will render MachineConfig
+manifests with higher precedence than the default ones created by the
+installer during bootstrapping. These manifests will include all of
+the details for configuring `kubelet`, CRI-O, and `tuned` to know
+about and partition workloads, including the cpusets for each workload
+type.
+
+The PAO `render` command also generates a manifest with the CRD for
+PerformanceProfile so bootstrapping does not block when trying to
+install the PerformanceProfile manifests.
+
+### Future Work
```

Comment:
any special instructions on how to address or pull out the pao binary for eventually invoke the render command?

Reply:
We do want to spend some time thinking about the UX there. Maybe that should happen in the PAO enhancement (item 2 below).

Reply:
Yep, this is something I touched upon in the PAO render mode enhancement. The model we want to use is to let the user execute it via `podman pullspec/pao:version <render-command-stuff>` from his laptop.